rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,521 @@
+            return self._decode_makerdao_close_cdp(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_BORROW_SAI_TOPIC:
+            return self._decode_makerdao_borrow_sai_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_REPAY_SAI_TOPIC:
+            return self._decode_makerdao_repay_sai_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_DEPOSIT_WETH_TOPIC:
+            return self._decode_makerdao_deposit_weth_eth_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_LIQUIDATE_TOPIC:
+            return self._decode_makerdao_liquidate_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_WITHDRAW_TOPIC:
+            return self._decode_makerdao_withdraw_eth(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+
+        return None, []
+
+    def _decode_makerdao_withdraw_eth(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        cdp_id = hex_or_bytes_to_int(tx_log.topics[2])
+        amount_withdrawn_raw = hex_or_bytes_to_int(tx_log.topics[3])
+        amount_withdrawn = asset_normalized_value(amount=amount_withdrawn_raw, asset=self.eth)
+
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount.is_close(amount_withdrawn, max_diff='0.003') and
```

Comment:
ooof. Why this .is_close? Why did you choose 0.003? There should be a comment explaining it (preferrably in the method's docstring). But it sounds wrong. Can't have accounting that works like this.

Reply:
The reason for this is that whenever a withdrawal or deposit occurs...

the amount deposited vs the collateral received isn't the same. Using the image below...
0.006 WETH was deposited and 0.0057 PETH was received/minted.

So it was a case of **use the values as is** or **use the 0.0057 value for both in & out event**
<img width="1040" alt="image" src="https://user-images.githubusercontent.com/72208758/207601394-2b133df3-291e-49fd-86d4-75bcc1f652b4.png">

Reply:
I responded inside the code to this. Resolving it since I have a different comment for this and the PR review did not let me continue the conversation here.